platform/src/main/java/org/hillview/sketches/ExactCountSketch.java

```diff
@@ -0,0 +1,80 @@
+package org.hillview.sketches;
+
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenCustomHashMap;
+import it.unimi.dsi.fastutil.objects.Object2IntOpenHashMap;
+import org.hillview.dataset.api.ISketch;
+import org.hillview.table.api.IRowIterator;
+import org.hillview.table.api.ITable;
+import org.hillview.table.rows.RowSnapshot;
+import org.hillview.table.rows.VirtualRowHashStrategy;
+import org.hillview.table.rows.VirtualRowSnapshot;
+import org.hillview.utils.MutableInteger;
+
+import javax.annotation.Nullable;
+import java.util.Arrays;
+
+public class ExactCountSketch implements ISketch<ITable, FreqKList> {
+    public CountSketchResult result;
+    public double threshold;
+    private final double cutoff;
+
+    public ExactCountSketch(CountSketchResult result, double threshold) {
+        this.result = result;
+        this.threshold = threshold;
+        this.cutoff =  this.threshold*this.result.estimateNorm();
+    }
+
+    @Override
+    public FreqKList create(ITable data) {
+        VirtualRowHashStrategy hashStrategy = new VirtualRowHashStrategy(data,
+                this.result.csDesc.schema);
+        Int2ObjectOpenCustomHashMap<MutableInteger> hMap =
+                new Int2ObjectOpenCustomHashMap<MutableInteger>(hashStrategy);
+        VirtualRowSnapshot vrs = new VirtualRowSnapshot(data, result.csDesc.schema);
+        IRowIterator rowIt = data.getRowIterator();
+        int i = rowIt.getNextRow();
+        MutableInteger val;
+        long item, hash;
+        int sign, toBucket;
+        long [] estimate = new long[this.result.csDesc.trials];
+        while (i != -1) {
+            val = hMap.get(i);
+            if (val != null) {
+                val.set(val.get() + 1);
+            } else {
+                vrs.setRow(i);
+                item = vrs.hashCode();
+                for (int j = 0; j < this.result.csDesc.trials; j++) {
+                    hash = this.result.csDesc.hashFunction[j].hashLong(item);
+                    sign = (hash % 2 == 0) ? 1 : -1;
+                    toBucket = (int) (Math.abs(hash / 2) % this.result.csDesc.buckets);
+                    estimate[j] = this.result.counts[j][toBucket] * sign;
+                }
+                Arrays.sort(estimate);
```

Comment:
You do one sort for each row of the data. That sounds expensive. If looks like you only need the median. There is a constant-time algorithm for median based on quicksort... Maybe it exists in a library.